### ISSUE TITLE
Array Initializer returns nullable?

### ISSUE THREAD

**Issue by NetTecture** [dotnet/roslyn#49316]
**Version Used**: 

5.0 RTM

**Steps to Reproduce**:

The following code is from a test framework putting together an oauth message. Note that the error is NOT related to the actual use, it is purely demonstrating:

            var nvc = new KeyValuePair<String, String>[]
            {
                new KeyValuePair<String, String>("grant_type", "refresh_token"),
                new KeyValuePair<String, String>("refresh_token", context.refresh_token)
            };

            var req = new HttpRequestMessage(HttpMethod.Post, "/token")
            {
                Content = new FormUrlEncodedContent(nvc)
            };

Note that this is the whole code using nvc. nvc is NOT ever touched again further in the method.

**Expected Behavior**:

No warnings shown. The variable nvc is defined as not nullable.

**Actual Behavior**:

The line new FormUrlEncodedContent(nvc) shows a warning:

Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS8620	Argument of type 'KeyValuePair<string, string>[]' cannot be used for parameter 'nameValueCollection' of type 'IEnumerable<KeyValuePair<string?, string?>>' in 'FormUrlEncodedContent.FormUrlEncodedContent(IEnumerable<KeyValuePair<string?, string?>> nameValueCollection)' due to differences in the nullability of reference types.	OpenDataCore.ServerTestSupport	C:\Work\CoreLibraries\OpenDataCore\Source\OpenDataCore.ServerTestSupport\TestSupport.cs	85	Active

The popup under the waning line also tells me "nvc is not null here".

The ultimate error seems to be that the first line of the example code:

var nvc = new KeyValuePair<String, String>[]

defines the var as nullable. This (local) variable is defined there and not never ever reused.

A current workaround is to tell the compiler it is not null, by changing the usage line to:

               Content = new FormUrlEncodedContent(nvc!)

(note the added !). This should not be needed.

**Comment by CyrusNajmabadi** [comment]
> defines the var as nullable. 

This is not the issue.  The compiler does think of htis as nullable, but knows it has a non-null state.  The issue you're seeing is that you have `KeyValuePair<string, string>` but the method you're calling expects `KeyValuePair<string?, string?>`s.  

You can likely fix this by doing:

```c#
        var nvc = new KeyValuePair<string?, string?>[]
        {
            new("grant_type", "refresh_token"),
            new("refresh_token", context.refresh_token)
        };
```
